comp/logs/agent/config/integration_config.go

```
@@ -76,22 +76,47 @@ type LogsConfig struct {
 	// ProcessRawMessage is used to process the raw message instead of only the content part of the message.
-	ProcessRawMessage *bool `mapstructure:"process_raw_message" json:"process_raw_message"`
+	ProcessRawMessage *bool `mapstructure:"process_raw_message" json:"process_raw_message" yaml:"process_raw_message"`
 
-	AutoMultiLine               *bool   `mapstructure:"auto_multi_line_detection" json:"auto_multi_line_detection"`
-	AutoMultiLineSampleSize     int     `mapstructure:"auto_multi_line_sample_size" json:"auto_multi_line_sample_size"`
-	AutoMultiLineMatchThreshold float64 `mapstructure:"auto_multi_line_match_threshold" json:"auto_multi_line_match_threshold"`
+	AutoMultiLine               *bool   `mapstructure:"auto_multi_line_detection" json:"auto_multi_line_detection" yaml:"auto_multi_line_detection"`
+	AutoMultiLineSampleSize     int     `mapstructure:"auto_multi_line_sample_size" json:"auto_multi_line_sample_size" yaml:"auto_multi_line_sample_size"`
+	AutoMultiLineMatchThreshold float64 `mapstructure:"auto_multi_line_match_threshold" json:"auto_multi_line_match_threshold" yaml:"auto_multi_line_match_threshold"`
+}
+
+// StringSliceField is a custom type for unmarshalling comma-separated string values or typical yaml fields into a slice of strings.
+type StringSliceField []string
+
+// UnmarshalYAML is a custom unmarshalling function is needed for string array fields to split comma-separated values.
+func (t *StringSliceField) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var str string
+	if err := unmarshal(&str); err == nil {
+		// note that we are intentionally avoiding the trimming of any spaces whilst splitting the string
+		str = strings.ReplaceAll(str, "\n", "")
+		*t = strings.Split(str, ",")
+		return nil
+	}
+
+	var raw []interface{}
+	if err := unmarshal(&raw); err == nil {
+		for _, item := range raw {
+			if str, ok := item.(string); ok {
+				*t = append(*t, str)
+			}
```

Should we return an error if the list contains non string element ?

---

Good idea, I added an error check, and modified a test to ensure that the error is raised.